### ISSUE TITLE
v2 remaining work breakdown

### ISSUE THREAD

**Issue by RubenSandwich** [NordicMuseum/Nordic-Museum-Audio-Guide#34]
### Sam
- [ ] Style tutorial
- [ ] New **Device** reducer which combines the reducers:
  - [ ] tutorial
  - [ ] accessibility
  - [ ] localization
  - [ ] nav
- [ ] Search by number working
  - [ ] Move number logic into reducer
  - [ ] Allow settings number of digits needed ()
- [ ] Styling nearMe (fake the data for now.)
- [ ] Remove unused code/assets/strings
- [ ] Android style fixes

### Ruben
- [ ] RTL bugs (v2.1?)
- [ ] Beacon support
- [ ] Splashscreen
- [ ] Reset to tutorial on charging
- [ ] Anayltics
- [ ] Tap title to go to active tour stop

### Aron
- [ ] iOS CD
- [ ] Android CD
- [ ] Writing new README

**Comment by Ambrosiani** [comment]
we're getting closer!

feedback from build 14, ios (doing android shortly):

- welcome image & tourstop images are not displayed
- tourstop headings are centered on the image, should be at bottom
- when museum mode is enabled, the app quits (crashes?) when a charger is unplugged. The tutorial part works as expected.
- in Arabic, some RTL changes still remain: number pad, play all button, bottom navigation, tourstop titles in tours.jsx
- 🎉 beacons are fully functioning! 🎉
- NearMe screen doesn't display the floor number
- NearMe displays highlights twice (this was a problem in the previous version as well for a while) – I think it is showing the highlight tracks both from the highlight tour and the Swedish Folk Art tour. I.e. how do we remove all highlights but the ones in the Highlight tour from the NearMe screen?
- search by number keypad is partly hidden by the bottomPlayer
- 🎉 amenities are fully functioning! 🎉

## already fixed
- I've added `ITSAppUsesNonExemptEncryption=NO` to XCode to get rid of the warning in TestFlight
- I've removed the build number from appVersion

## relevant screenshots
![IMG_7FB6479F2CE7-1](https://user-images.githubusercontent.com/2394405/65946334-37f9de80-e436-11e9-991d-d54b464d26cc.jpeg)
![IMG_4293](https://user-images.githubusercontent.com/2394405/65947761-3ed62080-e439-11e9-8e69-028d1a304244.PNG) ![IMG_4292](https://user-images.githubusercontent.com/2394405/65947763-3ed62080-e439-11e9-9e0e-d1e7790c690f.PNG) ![IMG_4291](https://user-images.githubusercontent.com/2394405/65947765-3ed62080-e439-11e9-8367-54ece7474be3.PNG)
![IMG_4296](https://user-images.githubusercontent.com/2394405/65948874-5d3d1b80-e43b-11e9-971c-dfcf4a9a5c80.PNG)
![IMG_4295](https://user-images.githubusercontent.com/2394405/65948875-5d3d1b80-e43b-11e9-8bae-f772a79e3326.PNG)
![IMG_4297](https://user-images.githubusercontent.com/2394405/65949322-43e89f00-e43c-11e9-8435-77fdac7e1144.PNG)

**Comment by RubenSandwich** [comment]
Aron can you add those bugs as a checklist on this issue?

**Comment by RubenSandwich** [comment]
@Ambrosiani I had some time and looked a bit into: " Compressing mp3s? Maybe move to aac? (80-100mb max.)" even though it was on your list. Even with medium quality aac I was only able to get the audio down to 730mb from 1gb. So I don't think the compression we need is even possible...

**Comment by Ambrosiani** [comment]
@RubenSandwich yeah, I did some experiments as well. Might be a good idea to switch anyway to bring down the app size, I'll find a good enough bitrate.

**Comment by sticknor** [comment]
@RubenSandwich 

I've completed Android style fixes with one exception. 
Maybe you've noticed when you switch locales, bottom tabs go white. Changing tab brings them back to OFF_BLACK.

I think this is a bug in react native navigation, since other people are filing similar issues. 
[Initial BottomTabs background color is always white](https://github.com/wix/react-native-navigation/issues/5198)
AND
[When using mergeOptions on bottomTabs, it loses it's background color](https://github.com/wix/react-native-navigation/issues/5533)

I’ve tried:
- Specifying background color again when Merging options for bottom bar
- Setting background defaults,
- Changing app theme in android manifest

All with no luck.
Gonna move to more pressing issues...
But wanted to record efforts here in case I can't get back to this.

**Comment by sticknor** [comment]
@RubenSandwich , I see one solution forward and with preliminary testing, it will work. If we can avoid merging options on locale change, we can avoid turning the bottom bar white. 

We can set the translated title in each container file instead of registerScreens. 
Should I proceed with this?

**Comment by RubenSandwich** [comment]
Yeah, try it. You can listen for the same event in each container.

One problem that might arise is how will we change the text of the
bottomBar during locale changes? Should we just reset the app for any
locale changes to avoid mergeOptions? (That might be easier.)

**Comment by sticknor** [comment]
@RubenSandwich Actually just removed that in hopes you hadn't see / worried about it yet... I was able to get it to work by specifying the bottomTabs background color on the screen type merge options. Redundant, and there's a small flicker,  but seems like the best case scenario.

You're right, not merging options will make the tab bar translation too difficult.

**Comment by RubenSandwich** [comment]
Haha no worries thanks for always being two steps ahead Sam. 💪🏽

**Comment by Ambrosiani** [comment]
@RubenSandwich @sticknor Sam is correct, the numpad should not change direction in RTL.

**Comment by RubenSandwich** [comment]
K, I think I just fixed most of the RTL bugs.

**Comment by Ambrosiani** [comment]
@RubenSandwich @sticknor I added an additional bug: previous audio doesn't stop when selecting another track

also my development build is running fine but the release build is crashing when a language is selected in the tutorial. I guess the release scheme needs some changes?

**Comment by RubenSandwich** [comment]
Okay, thanks Aron.

I will fix those tonight.
I will also email you an update shortly.

Ruben

**Comment by RubenSandwich** [comment]
@Ambrosiani I believe I have now fixed those bugs. Talk to you tomorrow.

**Comment by Ambrosiani** [comment]
@RubenSandwich awesome! the testflight build works as expected now.
